x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/TumblingWindow.java

```diff
@@ -918,4 +948,13 @@ public KeyAndOrdinal next() {
             };
         };
     }
+
+    private void addShardFailures(SearchResponse r) {
```

Comment:
I've seen this exact method in SampleIterator as well. Maybe extract it somewhere and re-use it?